tsl/src/compression/compression.c

```diff
@@ -2021,21 +2022,82 @@ is_segmentby_col(List *ht_compression_info, char *column_name)
 	return false;
 }
 
+/*
+ * Convert an expression to a Var referencing the index column.
+ * This method does following:
+ * 1. Change attribute numbers to match against index column position.
+ * 2. Set Var nodes varno to INDEX_VAR.
+ * 3. Reorder the Var nodes to align with column order in indexes.
+ *
+ * For details refer: match_clause_to_index() and fix_indexqual_operand()
+ */
+static void
+modify_attributes(Relation comp_chunk_rel, Var *var, List **pred, char *column_name, Node *node)
+{
+	Bitmapset *key_columns;
+	Relation index_rel;
+	List *index_oids;
+	Oid index_oid;
+	int i = 0;
+
+	key_columns = RelationGetIndexAttrBitmap(comp_chunk_rel, INDEX_ATTR_BITMAP_ALL);
+	index_oids = RelationGetIndexList(comp_chunk_rel);
+	index_oid = linitial_oid(index_oids);
```

Comment:
That is what i noticed. For a compressed chunk if there is a SEGMENT BY column defined, then there will be only 1 index. I will add an assert in case there is no index (in case someone drops this index :))